kclvm/compiler_base/3rdparty/rustc_errors/src/lib.rs

```diff
@@ -0,0 +1,362 @@
+//! Text rendering and related helper functions.
+//!
+//! Reuse 'styled_buffer.rs' in 'rustc_errors',
+//! and 'styled_buffer.rs' has been modified to fit the feature of 'Compiler-Base'.
+//!
+//! - add method 'appendl()' and 'putl()' to 'StyledBuffer'.
+//! - replaced the 'enum Style' with 'trait Style' to support extending more styles.
+//! - add some test cases for 'StyledBuffer'.
+
+use std::any::Any;
+use termcolor::ColorSpec;
+
+pub mod styled_buffer;
+
+/// 'Style' is a trait used to specify the 'enum Style' supported by 'StyleBuffer'.
```

Comment:
What is 'Style' is a trait used to specify the 'enum Style'?

Reply:
replaced the `enum Style` with generics `T: Clone + PartialEq + Eq + Style` to support extending more styles.
`StyledBuffer` still should be valid when facing the user-defined style, rather than just supporting a built-in `enum Style`.